basic-services/ReadPropertyResult.cs

```diff
@@ -39,42 +40,130 @@ public struct ReadPropertyResult
         // Helper 
         public bool IsReliable => Reliability == Reliable;
 
-        internal ReadPropertyResult(JToken token, CultureInfo cultureInfo, string reliability = null, string priority = null)
+        private CultureInfo _CultureInfo;
+
+        internal ReadPropertyResult(Guid id, JToken token, string attribute, CultureInfo cultureInfo = null, string reliability = null, string priority = null)
```

Comment:
I just discovered by looking at `ReadObject` that there is no need to pass in reliability and priority. If they are present, they will always be in the `JToken` object. So that will simplify the interface a bit.

Reply:
From my understanding these values will only be defined if the JToken is a JObject, which only happens if the attribute read is the PresentValue?

Reply:
Correct. I should have thought of that when suggesting the interface.